hugr-core/src/extension/simple_op.rs

```diff
@@ -138,10 +143,19 @@ impl<T: MakeOpDef> MakeExtensionOp for T {
 
 /// Load an [MakeOpDef] from its name.
 /// See [strum_macros::EnumString].
-pub fn try_from_name<T>(name: &OpNameRef) -> Result<T, OpLoadError>
+pub fn try_from_name<T>(def: &OpDef) -> Result<T, OpLoadError>
```

Comment:
`from_def` is abstract, `MakeOpDef` implementers need to have some way of loading themselves from a definition. It so happens that many can load themselves from (name, extension) because they implement `FromStr`, so we use a specific function for that case (`try_from_name`).

i dont mind passing name and extension explicitly like you suggest, so I've done that anyway. It's a bit clearer what the job of the function is.